server/models/testing/test-assignment.js

```diff
@@ -7,6 +7,7 @@ const testAssignmentSchema = new Schema({
   studentId: { type: Schema.Types.ObjectId, ref: 'User', index: true },
   name: { type: String, required: true },
   tags: { type: [{ type: String }], index: true, required: true },
+  sectionId: { type: Schema.Types.ObjectId, index: true, ref: 'Section' },

```

Comment:
There should be no sectionId. In this case you should fill Test-assignment's tags field with your section's tags.